### PR TITLE
grammar updates for AR::Persistence#reload

### PR DESCRIPTION
**Pull request by BenMorganIO**
Ran the sentences through [Grammarly](https://app.grammarly.com/) and used its suggestions. Happy to always improve docs, line them up to 80 characters, and ensure the ruby examples are 2 lines shorter :p

### CODE REVIEW

File: activerecord/lib/active_record/persistence.rb [hunk 1]
```diff
@@ -403,13 +403,14 @@ def toggle!(attribute)
     #   # => #<Account id: 1, email: 'account@example.com'>
     #
     # Attributes are reloaded from the database, and caches busted, in
-    # particular the associations cache and the QueryCache.
+    # particular, the association's cache and the QueryCache.
```
Comment (kaspth) [inline activerecord/lib/active_record/persistence.rb]: The old doc is correct. It's a cache of the associations, not a cache belonging to the associations. Comma change is fine 👍

File: activerecord/lib/active_record/persistence.rb [hunk 2]
```diff
@@ -421,20 +422,18 @@ def toggle!(attribute)
     #   assert_equal 25, account.credit        # check it is updated in memory
     #   assert_equal 25, account.reload.credit # check it is also persisted
     #
-    # Another common use case is optimistic locking handling:
+    # Another common use case is optimistic lock handling:
```
Comment (kaspth) [inline activerecord/lib/active_record/persistence.rb]: The feature is called `optimistic locking`, so it's better to use that phrasing exactly. Should be able to with:

```ruby
# Another common use case is handling optimistic locking:
```

**Review by kaspth — CHANGES_REQUESTED** [review]

**Comment by kaspth** [comment]
If you find legitimate grammar issues, do send PRs 👍

Less enthused about just submitting what some grammar site may find, we'd rather have human context 😉

File: activerecord/lib/active_record/persistence.rb (hunk 1 above)
Comment (matthewd) [inline activerecord/lib/active_record/persistence.rb]: I'd argue the comma change measurably exacerbates already-unclear grammar here.

Comment (kaspth) [inline activerecord/lib/active_record/persistence.rb]: Ah yeah, agreed 👍

**Comment by kaspth** [comment]
Applied the `finds the record` change in 8740c6e with credit, but left the other changes out. Thanks for the fix!

**Pull request closed without merging**